python-client/giskard/models/base/__init__.py

```diff
@@ -765,6 +768,18 @@ def load_model(cls, local_dir):
                 "serialisation method by overriding the save_model() and load_model() methods."
             )
 
+    def to_mlflow(self,
+                  artifact_path="prediction-function-from-giskard",
+                  **kwargs):
+        import mlflow
+
+        class MLflowModel(mlflow.pyfunc.PythonModel):
+            def predict(self, df):
+                return self.predict(df)
+
+        mlflow_model = MLflowModel()
+        return mlflow.pyfunc.log_model(artifact_path=artifact_path, python_model=mlflow_model)
```

Comment:
should we also log pre and post processing functions here? With `log_artefact`? Otherwise GIskard model will only be partially logged

Reply:
MLflow allows the custom creation of models if pre- or post-processing is needed, see: https://mlflow.org/docs/latest/python_api/mlflow.pyfunc.html?highlight=preprocessing. This means that if there's pre/post processing, it'll be already in the `predict()` of their pyfunc model, thus no need to log any additional functions from our side.